lib/generation/generators/value_objects/generation_configuration/pb_platform_orientation_generation_mixin.dart

```diff
@@ -1,42 +1,46 @@
+import 'package:parabeac_core/generation/generators/value_objects/file_structure_strategy/commands/node_file_structure_command.dart';
 import 'package:parabeac_core/generation/generators/value_objects/file_structure_strategy/commands/write_screen_command.dart';
 import 'package:parabeac_core/interpret_and_optimize/entities/subclasses/pb_intermediate_node.dart';
+import 'package:parabeac_core/interpret_and_optimize/helpers/pb_gen_cache.dart';
 import 'package:parabeac_core/interpret_and_optimize/helpers/pb_project.dart';
 import 'package:parabeac_core/interpret_and_optimize/services/pb_platform_orientation_linker_service.dart';
 import 'package:recase/recase.dart';
+import 'package:path/path.dart' as p;
 
 mixin PBPlatformOrientationGeneration {
-  void generatePlatformInstance(Map<String, List<String>> platformsMap,
-      String screenName, PBProject mainTree) {
-    platformsMap.forEach((platform, screens) {
-      var formatedName = screenName.snakeCase.toLowerCase();
-      if (screens.length > 1) {
-        _generateOrientationInstance(screenName, mainTree);
-      }
-      if (platformsMap.length > 1) {
-        mainTree.genProjectData.commandQueue.add(WriteScreenCommand(
-          formatedName + '_platform_builder.dart',
-          formatedName,
-          _getPlatformInstance(platformsMap, screenName),
+  NodeFileStructureCommand generatePlatformInstance(
+      Map<String, List<String>> platformsMap,
+      String screenName,
+      PBProject mainTree,
+      Set<String> rawImports) {
+    var formatedName = screenName.snakeCase.toLowerCase();
+    var cookedImports = _cookImports(
+        rawImports,
+        p.join(
```

Comment:
For the paths it should be
```dart
p.join(mainTree.fileStructureStrategy.GENERATED_PROJECT_PATH,
WriteScreenCommand.SCREEN_PATH,
'/$formatedName',
'/${formatedName}_plaform_builder.dart');
```
where instead of manually concatenating the paths manually, the `p.join(item, item, ...)` does it automatically.